test/test-buildURL.js

```diff
@@ -223,11 +223,31 @@ describe('URL Builder:', function describeSuite() {
             assert.equal(expectation, result);
         });
 
+        it('does not modify its input-argument', function testSpec() {
+            var params = {
+                    w: 400,
+                    h: 300
+                },
+                expectation = '?w=400&h=300',
+                result = client._buildParams(params);
+
+            assert.equal(params.w, 400);
+            assert.equal(params.h, 300);
+            assert.equal(expectation, result);
+
+            var emptyParams = {};
+            var emptyResult = client._buildParams(emptyParams);
+
+            // Ensure the result is empty.
+            assert.equal(emptyResult, '');
```

Comment:
"Ensure the result is empty" does not seem like a valid assertion given the name of the test - I suggest either to update the test name or drop the assertion (my opinion would be drop the assertion).

Reply:
Agreed. I think I left this in because it was interesting. I wasn't sure if this was documented behavior? I think this should go into a new test (or perhaps just dropped).